src/vma/dev/net_device_val.cpp

```diff
@@ -1675,7 +1675,7 @@ bool net_device_val::verify_qp_creation(const char* ifname, enum ibv_qp_type qp_
 	qp_init_attr.send_cq = cq;
 
 	// Set source qpn for mlx5 IPoIB devices
-	if (qp_type == IBV_QPT_UD && !strncmp(p_ib_ctx->get_ibname(), "mlx5", 4)) {
+	if (qp_type == IBV_QPT_UD && !p_ib_ctx->is_mlx4()) {
```

Comment:
It is dangerous way to check. Let imagine that there is more than just mlx4 and mlx5. Please consider `is_mlx5()` to be consistent.